plugins/experimental/ssl_cert_loader/ssl-cert-loader.cc

```diff
@@ -297,32 +296,40 @@ Parse_Config(Value &parent, ParsedSslValues &orig_values)
     std::deque<std::string> cert_names;
     SslEntry *entry = Load_Certificate_Entry(cur_values, cert_names);
 
-    // Store in appropriate table
-    if (cur_values.server_name.length() > 0) {
-      Lookup.tree.insert(cur_values.server_name, entry, Parse_order++);
-      inserted = true;
-    }
-    if (cur_values.server_ips.size() > 0) {
-      for (auto &server_ip : cur_values.server_ips) {
-        IpEndpoint first, second;
-        first.assign(server_ip.first);
-        second.assign(server_ip.second);
-        Lookup.ipmap.fill(&first, &second, entry);
-        char val1[256], val2[256];
-        server_ip.first.toString(val1, sizeof(val1));
-        server_ip.second.toString(val2, sizeof(val2));
+    if (entry) {
+      bool inserted = false;
+
+      // Store in appropriate table
+      if (cur_values.server_name.length() > 0) {
+        Lookup.tree.insert(cur_values.server_name, entry, Parse_order++);
         inserted = true;
       }
-    }
-    if (entry != nullptr) {
+      if (cur_values.server_ips.size() > 0) {
+        for (auto &server_ip : cur_values.server_ips) {
+          IpEndpoint first, second;
+          first.assign(server_ip.first);
+          second.assign(server_ip.second);
+          Lookup.ipmap.fill(&first, &second, entry);
+          char val1[256], val2[256];
+          server_ip.first.toString(val1, sizeof(val1));
+          server_ip.second.toString(val2, sizeof(val2));
```

Comment:
What purpose does it serve to fill in the block-local strings val1 and val2?

Reply:
Probably remnants of past debugging efforts.

Reply:
Yeh, I didn't look at any of that, just trying to address the coverity memory leak issue :).